### ISSUE TITLE
panic: runtime error: send on closed channel

### ISSUE THREAD

**Issue by kurtzhong** [bitly/go-nsq#144]
```
panic: runtime error: send on closed channel

goroutine 10818 [running]:
runtime.panic(0xd11480, 0x15748fe)
	/usr/local/go/src/pkg/runtime/panic.c:279 +0xf5
github.com/bitly/go-nsq.(*Consumer).onConnMessage(0xc208688000, 0xc209069200, 0xc208bba000)
	/root/gopath/src/mysite.com.cn/kefu/portal/Godeps/_workspace/src/github.com/bitly/go-nsq/consumer.go:632 +0x90
github.com/bitly/go-nsq.(*consumerConnDelegate).OnMessage(0xc208fa45e0, 0xc209069200, 0xc208bba000)
	/root/gopath/src/mysite.com.cn/kefu/portal/Godeps/_workspace/src/github.com/bitly/go-nsq/delegates.go:111 +0x3e
github.com/bitly/go-nsq.(*Conn).readLoop(0xc209069200)
	/root/gopath/src/mysite.com.cn/kefu/portal/Godeps/_workspace/src/github.com/bitly/go-nsq/conn.go:518 +0xb9b
created by github.com/bitly/go-nsq.(*Conn).Connect
	/root/gopath/src/mysite.com.cn/kefu/portal/Godeps/_workspace/src/github.com/bitly/go-nsq/conn.go:181 +0x6af

```

Could this be a issue in the go-nsq?

**Comment by mreiferson** [comment]
@kurtzhong can you provide more of the logs before the `panic` so I can try to figure out what happened?

**Comment by kurtzhong** [comment]
Unfortunately i don't have more relevant logs, and I don't meet this error in last two days. I found i was using the `1.0.4-alpha` version and I've update the package to `1.0.4`.

**Comment by mreiferson** [comment]
closing, assuming this has been fixed in `1.0.4+`
